File: step/stage10/go/fibonacci.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	fmt.Print(fib(n))
}

func fib(n int) int {
	if n < 2 {
		return n
	}

	return fib(n-2) + fib(n-1)
}

File: step/stage10/ts/fibonacci.ts
const stdin = process.openStdin()

stdin.addListener("data", (d: Buffer) => {
    const n = Number(d.toString())
    console.log(fib(n))
})

const fib = (n: number): number => {
    if(n < 2) {
        return n
    }

    return fib(n-2) + fib(n-1)
}